include/boost/multiprecision/cpp_int.hpp

```diff
@@ -231,11 +233,11 @@ struct cpp_int_base<MinBits, MaxBits, signed_magnitude, Checked, Allocator, fals
 
       constexpr data_type() noexcept : first(0) {}
       constexpr data_type(limb_type i) noexcept : first(i) {}
-      constexpr data_type(signed_limb_type i) noexcept : first(i < 0 ? static_cast<limb_type>(boost::multiprecision::detail::unsigned_abs(i)) : i) {}
+      constexpr data_type(signed_limb_type i) noexcept : first(i < 0 ? static_cast<limb_type>(boost::multiprecision::detail::unsigned_abs(i)) : static_cast<limb_type>(i)) {}
```

Comment:
I'm thinking we could simplify this, and just initialize with the result of `unsigned_abs(i)` which would handle both the `i < 0` check and the cast to limb_type for us?

Same with the next one below.

Reply:
Done